install/helm/agones/templates/_helpers.tpl

```diff
@@ -69,4 +69,24 @@ feature gates. In the process, we validate that the feature gate is known.
 {{- end -}}
 {{- end -}}
 {{- end -}}
-{{- end -}}
\ No newline at end of file
+{{- end -}}
+
+{{/* 
+Derives the replica number for agones-controller based on current helm and feature flag settings.
+*/}}
+{{- define "agones.controller.replica" -}}
+{{- $featureGates := include "agones.featureGates" . | fromYaml -}}
+{{- if not .Values.agones.controller.replicas -}}
+{{- if $featureGates.SplitControllerAndExtensions -}}
+{{- default 2 .Values.agones.controller.replicas -}}
+{{- else -}}
+{{- default 1 .Values.agones.controller.replicas -}}
+{{- end -}}
+{{- else -}}
+{{- if not $featureGates.SplitControllerAndExtensions -}}
+{{- default 1 .Values.agones.controller.replicas -}}
+{{- else -}}
+{{- .Values.agones.controller.replicas -}}
```

Comment:
I like this logic better than what I specified (and update the design!), but maybe we can strip it down even further:

* if SprintControllerAndExtensions, use the replicas values passed in
* if not, replicas=1

Basically, remove the line 79-84 clause, we don't need to protect against an empty value.

As a nit, I would flip the order of clause in line 86 and 88, though: in an if/else, prefer to keep the "positive" case first.